### PR TITLE
Mythological V2

### PR DESCRIPTION
**Pull request by SavageBishop**
#### Summary 
```SUMMARY: None```

#### Purpose of change
It was brought to my attention that orichalcum arrows couldn't be stored in quivers, so I made it happen. Plus allowed the Ichaival to fire regular arrow types.

#### Describe the solution
Quiver now supports orichalcum arrows, and the bow supports multiple ammo types.

#### Describe alternatives you've considered
Considered a new quiver special to the Ichaival, might still be a thing.

#### Testing
Added ammo type, to both quiver and Ichaival - spawned in fresh and loaded the quiver.

#### Additional context
As long as the ammo is defined as an arrow, the bow should fire it regardless of type. This might be changed but currently I don't see an issue so far.

### CODE REVIEW

**Comment by Maleclypse** [comment]
SUMMARY: Content  "Adds the ability to store orichalcum arrows in quivers, Ichaival can now fire normal arrows."  will fix your failing PR validator.  I'm not sure why your JSON validator is failing, I copied the armor file and ran it through http://dev.narc.ro/cataclysm/format.html and didn't see any immediate changes.

**Comment by SavageBishop** [comment]
> SUMMARY: Content "Adds the ability to store orichalcum arrows in quivers, Ichaival can now fire normal arrows." will fix your failing PR validator. I'm not sure why your JSON validator is failing, I copied the armor file and ran it through http://dev.narc.ro/cataclysm/format.html and didn't see any immediate changes.

Lmfao I tried that, and it still broke.

**Comment by tenmillimaster** [comment]
`Mods` as a summary tag might be more appropriate than `content`

**Comment by SavageBishop** [comment]
Hopefully this adds the boulder ( Which is needed to get orichalcum to even attempt to craft the items ) the regional settings. Hopefully it has the same spawn chances as the large boulder, unfortunately however I think they are identical in appearance.

File: data/mods/Magiclysm/worldgen/regional_overlay.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "type": "region_overlay",
     "//id": "magiclysm_buildings_overlay",
     "regions": [ "all" ],
-    "city": { "shops": { "magic_shop": 100, "used_bookstore": 225 }, "basements": { "magic_basement": 50 } }
+    "city": { "shops": { "magic_shop": 100, "used_bookstore": 225 }, "basements": { "magic_basement": 50 } },
+    "other": { "f_glow_boulder": 0.3333 }
```
Comment (ZhilkinSerg) [inline data/mods/Magiclysm/worldgen/regional_overlay.json]: That's incorrect - `other` node should be inside of `field_coverage` node:

```suggestion
    "field_coverage":  { "other": { "f_glow_boulder": 0.3333 } }
```

Comment (SavageBishop) [reply data/mods/Magiclysm/worldgen/regional_overlay.json]: How would I fix this, since it's already been merged?

Comment (ZhilkinSerg) [reply data/mods/Magiclysm/worldgen/regional_overlay.json]: In a separate PR like this - https://github.com/CleverRaven/Cataclysm-DDA/pull/35834

Comment (SavageBishop) [reply data/mods/Magiclysm/worldgen/regional_overlay.json]: Wait so you're already doing it? I'm confused.

**Pull request merged**